bookservice/src/main/java/com/epam/app/service/BookServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.epam.app.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.epam.app.dto.BookDto;
+import com.epam.app.model.Book;
+import com.epam.app.repository.BookRepository;
+
+@Service
+public class BookServiceImpl implements BookService{
+	
+	@Autowired
+	BookRepository bookRepository;
+
+	@Override
+	public Book add(BookDto bookDto) {
+		Book book = new Book(bookDto.getTitle(), bookDto.getAuthor(), bookDto.getCategory(), bookDto.getPublisher());
+		return bookRepository.save(book);
+	}
+
+	@Override
+	public List<Book> getAll() {
+		return bookRepository.findAll();
+	}
+
+	@Override
+	public Book get(int id) {
+		Optional<Book> book = bookRepository.findById(id);
+		if(book.isPresent())
+			return book.get();
+		return null;
```

Comment:
returning null is not a good practise, instead you can throw an exception saying book not found with given id